File: src/main/java/org/example/observer/stockmarket2/StockTrader.java
package org.example.observer.stockmarket2;


public class StockTrader implements Observer {

    private String traderName;

    public StockTrader(String traderName) {
        this.traderName = traderName;
    }

    @Override
    public void update(String stockSymbol, double stockPrice) {
        System.out.println(traderName + " received" + "new price : " + stockPrice + "for " + stockSymbol);
    }
}

File: kotlin/src/main/kotlin/observer/StockTrader.kt
package observer

class StockTrader:Observer  {

   lateinit var traderName:String
        private set

    public constructor(traderName: String) {
        this.traderName = traderName
    }


    override fun update(price: Int, product: String) {
        println(traderName+ "   "+ price + "  " + product)
    }
}